### PR TITLE
NEW: restore default settings (ISX-1564)

### PR DESCRIPTION
**Pull request by ritamerkl**
### Description

This PR adds a kebab menu to the global input project settings view with the option to reset. Find details in the [ticket](https://jira.unity3d.com/browse/ISX-1564).

### Changes made

added:
- kebab menu with option to reset
- reset global input settings to default logic

fixed: 
- fixed warnings for unused properties in ContextMenu

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs [hunk 1]
```diff
@@ -32,6 +32,11 @@ internal static void Reset()
 
 #endif
 
+        internal static InputActionAsset ResetAssetToDefault()
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: I would remove this function and just call `CreateNewActionAsset` directly.
From the context of `ProjectWideActionsAsset` nothing is being reset here.

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs [hunk 2]
```diff
@@ -12,14 +12,18 @@ internal class InputActionsEditorView : ViewBase<InputActionsEditorView.ViewStat
     {
         private const string saveButtonId = "save-asset-toolbar-button";
         private const string autoSaveToggleId = "auto-save-toolbar-toggle";
+        private const string menuButtonId = "asset-menu";
 
         Action m_PostSaveAction;
+        Action<InputActionAsset> m_PostResetAction;
 
-        public InputActionsEditorView(VisualElement root, StateContainer stateContainer, Action mpostSaveAction)
+        public InputActionsEditorView(VisualElement root, StateContainer stateContainer, Action mpostSaveAction, Action<InputActionAsset> postResetAction = null)
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs]: Is there any use-case for it to ever be left null?

I'm also concerned that adding callbacks here is not going to scale.

The mpostSaveAction also looks to be in the wrong place here (That should be in `InputActionsEditorWindowUtils.SaveAsset() directly IMO. Although that's not in this PR).

So I think the pattern should be something like:
`Dispatch(Commands.ResetAsset());`
->
`InputActionsEditorWindowUtils.ResetAsset();`

**Review by jamesmcgill — CHANGES_REQUESTED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs [hunk 3]
```diff
@@ -65,6 +78,12 @@ public void BuildUI()
                 });
         }
 
+        private void OnReset()
+        {
+            var asset = ProjectWideActionsAsset.ResetAssetToDefault();
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs]: I don't think this belongs in the View.
There is nothing else like it here (anything dealing with state).
I think you need to `Dispatch` a new Command from here to keep it consistent with the reset of the code.

**Comment by jamesmcgill** [comment]
Also, just make sure that `InputSystem.actions` (the actions used at runtime) also get reset too.
There is a sample `Assets/Samples/ProjectWideActions` to test it.

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs (hunk 2 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs]: I don't know if there will be a reset behavior coming up for the custom Input Action assets.. if so the callback would be different here - but now it's null. If something like that won't be upcoming I can easily remove it

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ however, it has to be formatted properly to pass verification tests.
 - Support for [Game rotation vector](https://developer.android.com/reference/android/hardware/Sensor#TYPE_GAME_ROTATION_VECTOR) sensor on Android
 - Duplicate Input Action Items in the new Input Action Asset Editor with Ctrl+D (Windows) or Cmd+D (Mac)
 - Selection of InputActionReferences from project-wide actions on fields that are of type InputActionReference. Uses a new advanced object picker that allows better searching and filtering of actions.
+- Reset project wide Input Settings to default via a new Kebab-menu in Input System Project Settings.
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md:29]: Is this an official menu name we are using?

Comment (jamesmcgill) [reply Packages/com.unity.inputsystem/CHANGELOG.md:29]: Kebab-menu is the standard 3 vertical dots type menu (as opposed to the hamburger-menu with 3 lines).

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs (hunk 2 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs]: changed it to an event now to (for both post save and reset)

**Comment by Pauliusd01** [comment]
Found a bug with PlayerInput ProjectWideActions dropdown option. Steps to repro (did not have much time so they are unreliable but hopefully the errors will help) : Go to the "SimpleDemo_UsingPlayerInput" in the main test project -> Select Player object and select ProjectWideActions in the PlayerInput component -> Reset ProjectWideActions in the ProjectSettings -> Select the object that had the PlayerInput component again -> errors are thrown: 

```
Action map must have state at this point
UnityEngine.Debug:ExtractStackTraceNoAlloc (byte*,int,string)
UnityEngine.StackTraceUtility:ExtractStackTrace () (at D:/Gitrepo/unity/Runtime/Export/Scripting/StackTrace.cs:37)
UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:Assert (bool,string)
UnityEngine.InputSystem.InputAction:get_currentState () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:1500)
UnityEngine.InputSystem.InputAction:get_phase () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:508)
UnityEngine.InputSystem.InputAction:get_enabled () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:533)
UnityEngine.InputSystem.InputActionAsset:FindAction (string,bool) (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputActionAsset.cs:551)
UnityEngine.InputSystem.Editor.PlayerInputEditor:OnActionAssetChange () (at ./Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs:448)
UnityEngine.InputSystem.Editor.PlayerInputEditor:OnInspectorGUI () (at ./Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs:79)
UnityEditor.UIElements.InspectorElement/<>c__DisplayClass74_0:<CreateInspectorElementUsingIMGUI>b__0 () (at D:/Gitrepo/unity/Editor/Mono/UIElements/Inspector/InspectorElement.cs:727)
UnityEngine.UIElements.IMGUIContainer:DoOnGUI (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,bool,UnityEngine.Rect,System.Action,bool) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:394)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,System.Action,bool) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:698)
UnityEngine.UIElements.IMGUIContainer:DoIMGUIRepaint () (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:574)
UnityEngine.UIElements.UIR.RenderChainCommand:ExecuteNonDrawMesh (UnityEngine.UIElements.UIR.DrawParams,single,System.Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderers.cs:134)
UnityEngine.UIElements.UIR.UIRenderDevice:EvaluateChain (UnityEngine.UIElements.UIR.RenderChainCommand,UnityEngine.Material,UnityEngine.Material,UnityEngine.Texture,UnityEngine.Texture,single,System.Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderDevice.cs:1017)
UnityEngine.UIElements.UIR.RenderChain:Render () (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRenderChain.cs:464)
UnityEngine.UIElements.UIRRepaintUpdater:Update () (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRepaintUpdater.cs:84)
UnityEngine.UIElements.VisualTreeUpdater:UpdateVisualTreePhase (UnityEngine.UIElements.VisualTreeUpdatePhase) (at D:/Gitrepo/unity/Modules/UIElements/Core/VisualTreeUpdater.cs:114)
UnityEngine.UIElements.Panel:UpdateForRepaint () (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1187)
UnityEngine.UIElements.Panel:Repaint (UnityEngine.Event) (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1251)
UnityEngine.UIElements.UIElementsUtility:DoDispatch (UnityEngine.UIElements.BaseVisualElementPanel) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:475)
UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent (int,intptr,bool&) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:225)
UnityEngine.UIElements.UIEventRegistration:ProcessEvent (int,intptr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:74)
UnityEngine.UIElements.UIEventRegistration/<>c:<.cctor>b__1_2 (int,intptr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:28)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at D:/Gitrepo/unity/Modules/IMGUI/GUIUtility.cs:206)
```

```
NullReferenceException: Object reference not set to an instance of an object
UnityEngine.InputSystem.InputAction.get_currentState () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:1501)
UnityEngine.InputSystem.InputAction.get_phase () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:508)
UnityEngine.InputSystem.InputAction.get_enabled () (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs:533)
UnityEngine.InputSystem.InputActionAsset.FindAction (System.String actionNameOrId, System.Boolean throwIfNotFound) (at ./Packages/com.unity.inputsystem/InputSystem/Actions/InputActionAsset.cs:551)
UnityEngine.InputSystem.Editor.PlayerInputEditor.OnActionAssetChange () (at ./Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs:448)
UnityEngine.InputSystem.Editor.PlayerInputEditor.OnInspectorGUI () (at ./Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs:79)
UnityEditor.UIElements.InspectorElement+<>c__DisplayClass74_0.<CreateInspectorElementUsingIMGUI>b__0 () (at D:/Gitrepo/unity/Editor/Mono/UIElements/Inspector/InspectorElement.cs:727)
UnityEngine.DebugLogHandler:Internal_LogException_Injected(Exception, IntPtr)
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEditor.UIElements.<>c__DisplayClass74_0:<CreateInspectorElementUsingIMGUI>b__0() (at D:/Gitrepo/unity/Editor/Mono/UIElements/Inspector/InspectorElement.cs:738)
UnityEngine.UIElements.IMGUIContainer:DoOnGUI(Event, Matrix4x4, Rect, Boolean, Rect, Action, Boolean) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:394)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent(Event, Matrix4x4, Rect, Action, Boolean) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:698)
UnityEngine.UIElements.IMGUIContainer:DoIMGUIRepaint() (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:574)
UnityEngine.UIElements.UIR.RenderChainCommand:ExecuteNonDrawMesh(DrawParams, Single, Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderers.cs:134)
UnityEngine.UIElements.UIR.UIRenderDevice:EvaluateChain(RenderChainCommand, Material, Material, Texture, Texture, Single, Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderDevice.cs:1017)
UnityEngine.UIElements.UIR.RenderChain:Render() (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRenderChain.cs:464)
UnityEngine.UIElements.UIRRepaintUpdater:Update() (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRepaintUpdater.cs:84)
UnityEngine.UIElements.VisualTreeUpdater:UpdateVisualTreePhase(VisualTreeUpdatePhase) (at D:/Gitrepo/unity/Modules/UIElements/Core/VisualTreeUpdater.cs:114)
UnityEngine.UIElements.Panel:UpdateForRepaint() (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1187)
UnityEngine.UIElements.Panel:Repaint(Event) (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1251)
UnityEngine.UIElements.UIElementsUtility:DoDispatch(BaseVisualElementPanel) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:475)
UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent(Int32, IntPtr, Boolean&) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:225)
UnityEngine.UIElements.UIEventRegistration:ProcessEvent(Int32, IntPtr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:74)
UnityEngine.UIElements.<>c:<.cctor>b__1_2(Int32, IntPtr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:28)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&) (at D:/Gitrepo/unity/Modules/IMGUI/GUIUtility.cs:206)
```

```
ArgumentException: Getting control 2's position in a group with only 2 controls when doing repaint
Aborting
UnityEngine.GUILayoutGroup.GetNext () (at D:/Gitrepo/unity/Modules/IMGUI/LayoutGroup.cs:129)
UnityEngine.GUILayoutUtility.DoGetRect (System.Single minWidth, System.Single maxWidth, System.Single minHeight, System.Single maxHeight, UnityEngine.GUIStyle style, UnityEngine.GUILayoutOption[] options) (at D:/Gitrepo/unity/Modules/IMGUI/GUILayoutUtility.cs:499)
UnityEngine.GUILayoutUtility.GetRect (System.Single minWidth, System.Single maxWidth, System.Single minHeight, System.Single maxHeight, UnityEngine.GUIStyle style, UnityEngine.GUILayoutOption[] options) (at D:/Gitrepo/unity/Modules/IMGUI/GUILayoutUtility.cs:488)
UnityEditor.EditorGUILayout.GetControlRect (System.Boolean hasLabel, System.Single height, UnityEngine.GUIStyle style, UnityEngine.GUILayoutOption[] options) (at D:/Gitrepo/unity/Editor/Mono/EditorGUILayout.cs:2118)
UnityEditor.EditorGUILayout.Popup (UnityEngine.GUIContent label, System.Int32 selectedIndex, UnityEngine.GUIContent[] displayedOptions, UnityEngine.GUIStyle style, UnityEngine.GUILayoutOption[] options) (at D:/Gitrepo/unity/Editor/Mono/EditorGUILayout.cs:926)
UnityEditor.EditorGUILayout.Popup (UnityEngine.GUIContent label, System.Int32 selectedIndex, UnityEngine.GUIContent[] displayedOptions, UnityEngine.GUILayoutOption[] options) (at D:/Gitrepo/unity/Editor/Mono/EditorGUILayout.cs:920)
UnityEngine.InputSystem.Editor.PlayerInputEditor.OnInspectorGUI () (at ./Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs:88)
UnityEditor.UIElements.InspectorElement+<>c__DisplayClass74_0.<CreateInspectorElementUsingIMGUI>b__0 () (at D:/Gitrepo/unity/Editor/Mono/UIElements/Inspector/InspectorElement.cs:727)
UnityEngine.DebugLogHandler:Internal_LogException_Injected(Exception, IntPtr)
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEditor.UIElements.<>c__DisplayClass74_0:<CreateInspectorElementUsingIMGUI>b__0() (at D:/Gitrepo/unity/Editor/Mono/UIElements/Inspector/InspectorElement.cs:738)
UnityEngine.UIElements.IMGUIContainer:DoOnGUI(Event, Matrix4x4, Rect, Boolean, Rect, Action, Boolean) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:394)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent(Event, Matrix4x4, Rect, Action, Boolean) (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:709)
UnityEngine.UIElements.IMGUIContainer:DoIMGUIRepaint() (at D:/Gitrepo/unity/Modules/UIElements/Core/IMGUIContainer.cs:574)
UnityEngine.UIElements.UIR.RenderChainCommand:ExecuteNonDrawMesh(DrawParams, Single, Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderers.cs:134)
UnityEngine.UIElements.UIR.UIRenderDevice:EvaluateChain(RenderChainCommand, Material, Material, Texture, Texture, Single, Exception&) (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRenderer/UIRenderDevice.cs:1017)
UnityEngine.UIElements.UIR.RenderChain:Render() (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRenderChain.cs:464)
UnityEngine.UIElements.UIRRepaintUpdater:Update() (at D:/Gitrepo/unity/Modules/UIElements/Core/Renderer/UIRRepaintUpdater.cs:84)
UnityEngine.UIElements.VisualTreeUpdater:UpdateVisualTreePhase(VisualTreeUpdatePhase) (at D:/Gitrepo/unity/Modules/UIElements/Core/VisualTreeUpdater.cs:114)
UnityEngine.UIElements.Panel:UpdateForRepaint() (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1187)
UnityEngine.UIElements.Panel:Repaint(Event) (at D:/Gitrepo/unity/Modules/UIElements/Core/Panel.cs:1251)
UnityEngine.UIElements.UIElementsUtility:DoDispatch(BaseVisualElementPanel) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:475)
UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent(Int32, IntPtr, Boolean&) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:225)
UnityEngine.UIElements.UIEventRegistration:ProcessEvent(Int32, IntPtr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:74)
UnityEngine.UIElements.<>c:<.cctor>b__1_2(Int32, IntPtr) (at D:/Gitrepo/unity/Modules/UIElements/Core/UIElementsUtility.cs:28)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&) (at D:/Gitrepo/unity/Modules/IMGUI/GUIUtility.cs:206)

```

Up to you if this sounds serious enough to be a blocker

**Review by Pauliusd01 — APPROVED** [review]
Found only one bug in my quick check. Things checked: whether PlayerInput still finds the newly reset project wide actions, whether input system settings are ignored by the resetting, whether any things remain unchanged after reset such as renames, rebinds ,etc.

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 4 above)
Comment (Pauliusd01) [reply Packages/com.unity.inputsystem/CHANGELOG.md:29]: Kebab 👍

**Review by jamesmcgill — APPROVED** [review]

**Comment by ritamerkl** [comment]
Good catch @Pauliusd01. I can reproduce this by just deleting an action and not resetting as well so it's not related to this PR. Anyway I think this should be filed

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs (hunk 2 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs]: continued here: https://jira.unity3d.com/browse/ISX-1728

**Pull request merged**